src/documents/models.py

```diff
@@ -398,13 +398,18 @@ def set_filename(self, filename):
             self.filename = filename
 
 
-def delete_empty_directory(directory):
-    if len(os.listdir(directory)) == 0:
+def try_delete_empty_directories(directory):
+    # Go up in the directory hierarchy and try to delete all directories
+    while directory != Document.filename_to_path(""):
+        # Try to delete the current directory
         try:
             os.rmdir(directory)
         except os.error:
-            # Directory not empty
-            pass
+            # Directory not empty, no need to go further up
+            return
+
+        # Cut off actual directory and go one level up
+        directory, tmp = os.path.split(directory)
```

Comment:
I think it is more established to name an unused/ignored value a single underscore, but this is a very minor nit-pick. :slightly_smiling_face: 

```suggestion
        directory, _ = os.path.split(directory)
```